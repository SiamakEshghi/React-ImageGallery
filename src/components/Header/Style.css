
.header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 50px;
    background-color: rgb(190, 121, 71);
}

.logo {
    max-width: 100px;
    height: auto;
    margin-left: 10px;
}
.loginContainer{
    display: flex;
    justify-content: center;
    width: 150px;
    margin-right: 10px;
    border-radius: 3px;
}
.loadingBtn {
    background-color: rgb(241, 197, 147);
    border-color: rgb(233, 138, 30);
    border-radius: 5px;
}
.linkContainer {
    width: 150px;
    display: flex;
    justify-content: space-between;
    margin-left: 10px;
}
.link{
    color: black;
    font-size: 12px;
}