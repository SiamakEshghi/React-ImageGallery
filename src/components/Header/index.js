import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import 'components/Header/Style.css';
import logo from 'img/logo.png'
import * as action from 'action';
 
 const Header = (props) => {
    const { isLogged } = props;

    const handleLogin = () => {
        props.changeLog(!isLogged);
     }
     return (
         <div className="header">
            <div className="linkContainer">
                <Link className="link" to="/" >Album</Link>
                <Link className="link" to="/upload" >Upload Photo</Link>
            </div>
            <img className="logo" src={logo} alt=""/>
            <div className="loginContainer">
                <button 
                 className="loadingBtn"
                 onClick={handleLogin}
                >{isLogged ? "Logout" : "Login"}</button>
            </div>
         </div>
        );    
 }
 const mapStateToProps = ({ auth }) => {
    return { isLogged: auth.isLogged }
 };

 export default connect(mapStateToProps, action)(Header);