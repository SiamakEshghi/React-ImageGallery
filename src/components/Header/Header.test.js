import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import reduxThunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import Header from 'components/Header';

let wrapper, store, parentWrapper;
const middlewares = [reduxThunk];
const mockStore = configureStore(middlewares);
const initialState = {
    img: { imageList: [{id:1, largeImageURL: 'url1'}]}, 
    auth: { isLogged: false }
};


beforeEach(() => {
    store = mockStore(initialState);
    parentWrapper = shallow(<Header store={store} /> );
    wrapper = parentWrapper.dive();
});

describe('Header UI', () => {

    it('should render correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('check isLogged Prop matches to isLogged initialState', () => {
        expect(parentWrapper.find('Header').prop('isLogged')).toEqual(initialState.auth.isLogged)
    });
});

describe('Login Button', () => {
    beforeEach(() => {
        wrapper.find('.loadingBtn').simulate('click');
        wrapper.update();
    });

    it('login button change Header isLogged Prop ', () => {
        setTimeout(() => expect(parentWrapper.find('Header').prop('isLogged')).toEqual(true),
        200);
    });

    it('login button change button lable ', () => {
        setTimeout(() => expect(wrapper.find('.loadingBtn').text()).toEqual('Logout'),
        200);
    });
    
});
