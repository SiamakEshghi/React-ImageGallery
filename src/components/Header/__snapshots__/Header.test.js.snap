// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header UI should render correctly 1`] = `
<div
  className="header"
>
  <div
    className="linkContainer"
  >
    <Link
      className="link"
      replace={false}
      to="/"
    >
      Album
    </Link>
    <Link
      className="link"
      replace={false}
      to="/upload"
    >
      Upload Photo
    </Link>
  </div>
  <img
    alt=""
    className="logo"
    src="logo.png"
  />
  <div
    className="loginContainer"
  >
    <button
      className="loadingBtn"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
